### DISCUSSION TITLE
wstring support

### DISCUSSION THREAD

**Discussion by Leadwerks** [nlohmann/json#2500] [Ideas]
Something like this would be nice:
```c++
	void JSONSetString(nlohmann::json& j3, const WString& s)
	{
		auto ss = s.ToString();
		if (s == ss)
		{
			j3 = ss;
		}
		else
		{
			j3 = s;
		}
	}

	WString JSONGetString(nlohmann::json& j3)
	{
		if (j3.is_string()) return stds::string(j3);
		if (j3.is_array())
		{
			std::wstring s;
			s.reserve(j3.size());
			for (int n = 0; n < j3.size(); ++n)
			{
				if (!j3[n].is_number_integer()) return L"";
				s.push_back(j3[n]);
			}
			return s;
		}
		return L"";
	}
```

**Comment by nlohmann** [comment]
The library is deliberately UTF-8-centric (JSON text is UTF-8 per RFC 8259), so `std::wstring` (UTF-16 or UTF-32 depending on platform) isn't supported as a first-class `StringType` out of the box — its width and encoding are platform-dependent (`wchar_t` is 16 bits on Windows, 32 on Linux/macOS), which doesn't map cleanly onto the library's UTF-8 validation logic. It's technically possible to use `std::wstring` as a custom `StringType` template parameter, but round-tripping still requires manual UTF-8 conversion at the boundary, which is likely why this hasn't been built in — this looks more like a deliberate design trade-off than an oversight.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
